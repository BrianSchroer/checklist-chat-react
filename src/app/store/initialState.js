import * as modalDialogType from '../modalDialogType';

export default {
    rooms: [],
    currentRoomId: null,
    ajaxCallsInProgressCount: 0,
    modalDialog: modalDialogType.NONE
};
