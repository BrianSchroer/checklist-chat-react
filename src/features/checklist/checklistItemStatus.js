export const NOT_STARTED                = 'NotStarted';
export const IN_PROGRESS                = 'InProgress';
export const IN_PROGRESS_WITH_ISSUES    = 'InProgressWithIssues';
export const COMPLETED_SUCCESSFULLY     = 'CompletedSuccessfully';
export const COMPLETED_WITH_ISSUES      = 'CompletedWithIssues';
export const COMPLETED_WITH_ERRORS      = 'CompletedWithErrors';
export const CANCELED                   = 'Canceled';
