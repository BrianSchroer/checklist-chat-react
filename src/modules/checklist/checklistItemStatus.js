export const NOT_STARTED                = 'NotStarted';
export const IN_PROGRESS                = 'InProgress';
export const IN_PROGRESS_WITH_ISSUES    = 'InProgressWithIssues';
export const COMPLETED_SUCCESSFULLY     = 'CompletedSuccessfully';
export const COMPLETED_WITH_ISSUES      = 'CompletedWithIssues';
export const COMPLETED_WITH_ERRORS      = 'CompletedWithErrors';
export const CANCELED                   = 'Canceled';

export const options = [
    { value: NOT_STARTED, text: 'Not Started' },
    { value: IN_PROGRESS, text: 'In Progress' },
    { value: IN_PROGRESS_WITH_ISSUES, text: 'In Progress With Issues' },
    { value: COMPLETED_SUCCESSFULLY, text: 'Completed Successfully' },
    { value: COMPLETED_WITH_ISSUES, text: 'Completed with Issues' },
    { value: COMPLETED_WITH_ERRORS, text: 'Completed with Errors' },
    { value: CANCELED, text: 'Canceled' }
];
