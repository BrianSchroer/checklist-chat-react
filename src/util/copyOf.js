export default original => Object.assign({}, original);
