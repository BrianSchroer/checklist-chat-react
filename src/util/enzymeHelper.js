const enzymeHelper = class {

    static nestedFind(wrapper, selector) {
        const selectors = selector.split('>').map(node => node.trim());
        let parent = wrapper;
        let found = [];

        for (let i = 0; i < selectors.length; i++) {
            found = parent.find(selectors[i]);
            if (!found.length) {
                break;
            }
        }

        return found;
    }

    static findSingle(wrapper, selector) {
        const findArray = enzymeHelper.nestedFind(wrapper, selector);

        if (findArray.length != 1) {
            throw(`expected 1 find for selector "${selector}", but found ${findArray.length}.`);
        }

        return findArray.first();
    }
};

export default enzymeHelper;
