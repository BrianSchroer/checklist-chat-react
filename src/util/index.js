import format from './format';
import uiHelper from './uiHelper';
import validationHelper from './validationHelper';

export {
    format as format,
    uiHelper as uiHelper,
    validationHelper as validationHelper
};
