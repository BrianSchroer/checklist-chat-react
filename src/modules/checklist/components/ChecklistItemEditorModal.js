import React, {PropTypes} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {saveChecklistItem} from '../checklistItemDucks';
import {requestChecklistItemCommentModalDialog} from '../../../app/modalDialogDucks';
import * as checklistItemStatus from '../checklistItemStatus';
import {validate} from '../checklistItemValidator';
import {FormGroup, Modal, SelectInput, TextInput, SimpleSelectInput} from '../../../modules/core';
import format from '../../../util/format';
import ChecklistItemStatusIcon from './ChecklistItemStatusIcon';

export class ChecklistItemEditorModal extends React.Component {
    constructor(props, context) {
        super(props, context);

        this.state = {
            checklistItem: Object.assign({}, props.checklistItem),
            isSaving: false,
            isDeleting: false,
            isDirty: false,
            comment: '',
            errors: {}
        };

        this.onChange = this.onChange.bind(this);
        this.onSave = this.onSave.bind(this);
        this.onCommentRequest = this.onCommentRequest.bind(this);
    }

    onChange(event) {
        const checklistItem = this.state.checklistItem;
        checklistItem[event.target.name] = event.target.value;
        return this.setState({checklistItem: checklistItem, isDirty: true});
    }

    onSave(event) {
        event.preventDefault();
        const {checklistItem} = this.state;

        const validationResponse = validate(checklistItem);

        if (!validationResponse.isValid) {
            this.setState({errors: validationResponse.errors});
            return;
        }

        const {userId, actions} = this.props;
        actions.saveChecklistItem(checklistItem, checklistItem.roomId, userId);
        this.props.onCloseRequest(event);
    }

    onCommentRequest(event) {
        event.preventDefault();

        const {actions, checklistItem} = this.props;

        actions.requestChecklistItemCommentModalDialog(
            checklistItem.roomId, checklistItem.sequenceNumber);
    }

    render() {
        const {isNewChecklistItem, maxSequenceNumber, onCloseRequest} = this.props;
        const {checklistItem, errors} = this.state;

        const sequenceNumberOptions = [...Array(maxSequenceNumber).keys()].map(i =>
        {
            const sequenceNumber = (i + 1).toString();
            return {value: sequenceNumber, text: sequenceNumber};
        });

        const buttons = (
            <div>
                {!isNewChecklistItem &&
                    <input type="button" value="Add Comment..."
                        className="btn btn-secondary pull-left" onClick={this.onCommentRequest} />
                }
                <input type="button" value="Cancel" className="btn btn-default"
                    onClick={onCloseRequest}/>
                <input type="submit" value="Save" className="btn btn-primary"
                    onClick={this.onSave}/>
            </div>
        );

        return (
            <Modal
                title={(isNewChecklistItem) ? 'Add New Checklist Item' : 'Edit Checklist Item'}
                onCloseRequest={onCloseRequest}
                buttons={buttons}>

                <div>
                    <div className="checklist-form-sequence-number-row">
                        <SelectInput name="sequenceNumber" label="Sequence Number"
                            options={sequenceNumberOptions}
                            value={checklistItem.sequenceNumber.toString()}
                            onChange={this.onChange} error={errors.sequenceNumber} />
                    </div>

                    <div className="checklist-form-status-row">
                        <FormGroup name="status" label="Status" error={errors.status}>
                            <div className="checklist-form-status-and-dropdown">
                                    <div className="form-control checklist-form-status-div">
                                    <ChecklistItemStatusIcon status={checklistItem.status}/>
                                </div>
                                <SimpleSelectInput
                                    name="status"
                                    value={checklistItem.status}
                                    options={checklistItemStatus.options}
                                    onChange={this.onChange} />
                            </div>
                        </FormGroup>
                    </div>

                    <TextInput name="description" label="Description" value={checklistItem.description}
                        rows={2} onChange={this.onChange} error={errors.description} />

                    <div className="checklist-form-timestamps">
                        <TextInput name="scheduledStartTime" label="Start Time: Scheduled"
                            value={format.time(checklistItem.scheduledStartTime)}
                            onChange={this.onChange} error={errors.scheduledStartTime} />
                        <TextInput name="actualStartTime" label="Actual"
                            value={format.time(checklistItem.actualStartTime)}
                            onChange={this.onChange} error={errors.actualStartTime} />
                    </div>

                    <div className="checklist-form-timestamps">
                        <TextInput name="scheduledEndTime" label="End Time: Scheduled"
                            value={format.time(checklistItem.scheduledEndTime)}
                            onChange={this.onChange} error={errors.scheduledEndTime} />
                        <TextInput name="actualEndTime" label="Actual"
                            value={format.time(checklistItem.actualEndTime)}
                            onChange={this.onChange} error={errors.actualEndTime} />
                    </div>

                    <TextInput name="userName" label="Performed by" value={checklistItem.userName}
                        onChange={this.onChange} error={errors.userName} />
                </div>

            </Modal>
        );
    }
}

ChecklistItemEditorModal.propTypes = {
    userId: PropTypes.string.isRequired,
    checklistItem: PropTypes.object.isRequired,
    isNewChecklistItem: PropTypes.bool.isRequired,
    maxSequenceNumber: PropTypes.number.isRequired,
    onCloseRequest: PropTypes.func.isRequired,
    actions: PropTypes.object.isRequired
};

function emptyChecklistItem(roomId, sequenceNumber) {
    return {
        id: null,
        roomId: roomId,
        sequenceNumber: sequenceNumber,
        status: checklistItemStatus.NOT_STARTED,
        description: null,
        scheduledStartTime: null,
        scheduledEndTime: null,
        actualStartTime: null,
        actualEndTime: null
    };
}

function mapStateToProps(state, ownProps) {
    let isNewChecklistItem = true;
    let maxSequenceNumber = 1;
    let checklistItem = {};

    const [roomId, sequenceNumber] = state.modalDialogRequest.keys;
    maxSequenceNumber = state.checklistItems.length + 1;

    if (sequenceNumber) {
        isNewChecklistItem = false;
        const checklistItems = state.checklistItems;
        checklistItem = checklistItems.find(item =>
            item.roomId === roomId && item.sequenceNumber === sequenceNumber);
        maxSequenceNumber--;
    } else {
        checklistItem = emptyChecklistItem(roomId, maxSequenceNumber);
    }

    const {userId} = state;
    const onCloseRequest = ownProps.onCloseRequest;

    return {userId, checklistItem, isNewChecklistItem, maxSequenceNumber, onCloseRequest};
}

const mapDispatchToProps = (dispatch) => ({
    actions: bindActionCreators(
        {saveChecklistItem, requestChecklistItemCommentModalDialog},
        dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(ChecklistItemEditorModal);
