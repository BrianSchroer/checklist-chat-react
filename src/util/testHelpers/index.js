import {shallow, mount} from 'enzyme';
import enzymeHelper from './enzymeHelper';

export {
    shallow as shallow,
    mount as mount,
    enzymeHelper as enzymeHelper
};
