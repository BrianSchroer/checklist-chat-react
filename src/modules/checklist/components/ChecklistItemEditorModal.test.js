import React from 'react';
import initialState from '../../../app/store/initialState';
import {snapshotHelper} from '../../../util/testHelpers';
import {ChecklistItemEditorModal, emptyChecklistItem,  mapStateToProps} from './ChecklistItemEditorModal';
import * as checklistItemStatus from '../checklistItemStatus';

const defaultProps = {
    checklistItem: {
        roomId: 123,
        sequenceNumber: 1,
        status: checklistItemStatus.NOT_STARTED,
        description: "item description",
        scheduledStartTime: '2016-12-08T14:00:00.222Z',
        scheduledEndTime: '2016-12-08T15:00:00.222Z',
        actualStartTime: '2016-12-08T14:03:00.222Z',
        actualEndTime: '2016-12-08T15:30:00.222Z',
        userName: 'Jabba Script' },
    isNewChecklistItem: false,
    maxSequenceNumber: 3,
    onCloseRequest: () => {},
    actions: {}
};

const defaultOwnProps = {
    onCloseRequest: () => {}
};

function overrideProps(propOverrides) {
    return Object.assign({}, defaultProps, propOverrides);
}

function assertSnapshotMatch(propOverrides = {}) {
    snapshotHelper.assertMatch(<ChecklistItemEditorModal {...overrideProps(propOverrides)} />);
}

function callMapStateToProps(stateOverrides, ownPropsOverrides) {
    const state = Object.assign({}, initialState, stateOverrides || {});
    const ownProps = Object.assign({}, defaultOwnProps, ownPropsOverrides || {});

    return mapStateToProps(state, ownProps);
}

describe('ChecklistItemEditorModal', () => {
    describe('for existing item', () => {
        it('should render correctly', () => {
            assertSnapshotMatch();
        });
    });

    describe('for new item', () => {
        it('should render correctly', () => {
            assertSnapshotMatch({ checklistItem: emptyChecklistItem(1, 3) });
        });
    });

    describe('mapStateToProps', () => {
        it('should return expected props for existing item', () => {
            const testChecklistItem = defaultProps.checklistItem;

            const stateOverrides = {
                modalDialogRequest: {
                     keys: [ testChecklistItem.roomId, testChecklistItem.sequenceNumber ]
                },
                checklistItems: [testChecklistItem]
            };

            const props = callMapStateToProps(stateOverrides);

            expect(props.checklistItem).toBe(testChecklistItem);
            expect(props.isNewChecklistItem).toBe(false);
            expect(props.maxSequenceNumber).toBe(testChecklistItem.sequenceNumber);
            expect(props.onCloseRequest).toBe(defaultOwnProps.onCloseRequest);
        });

        it('should return expected props for new item', () => {
            const testRoomId = defaultProps.checklistItem.roomId;

            const stateOverrides = {
                modalDialogRequest: { keys: [ testRoomId, 0 ] },
                checklistItems: [defaultProps.checklistItem]
            };

            const props = callMapStateToProps(stateOverrides);

            expect(props.checklistItem.roomId).toBe(testRoomId);
            expect(props.checklistItem.sequenceNumber).toBe(2);
            expect(props.isNewChecklistItem).toBe(true);
            expect(props.maxSequenceNumber).toBe(2);
            expect(props.onCloseRequest).toBe(defaultOwnProps.onCloseRequest);
        });
    });
});
