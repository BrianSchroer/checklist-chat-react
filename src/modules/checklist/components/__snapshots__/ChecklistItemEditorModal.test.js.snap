// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChecklistItemEditorModal for existing item should render correctly 1`] = `
<div
  className="checklist-chat-modal-backdrop"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog checklist-chat-modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="modal-header bg-primary text-white"
        >
          <button
            aria-label="Close"
            className="close"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
          <h4
            className="modal-title display-linebreaks"
          >
            Edit Checklist Item
          </h4>
        </div>
        <div
          className="modal-body"
        >
          <div>
            <div
              className="checklist-form-sequence-and-status-row"
            >
              <div
                className="form-group"
              >
                <label
                  htmlFor="sequenceNumber"
                >
                  Seq #
                </label>
                <div
                  className="field"
                >
                  <select
                    className="form-control"
                    name="sequenceNumber"
                    onChange={[Function]}
                    value="1"
                  >
                    <option
                      value="1"
                    >
                      1
                    </option>
                    <option
                      value="2"
                    >
                      2
                    </option>
                    <option
                      value="3"
                    >
                      3
                    </option>
                  </select>
                </div>
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="status"
                >
                  Status
                </label>
                <div
                  className="field"
                >
                  <div
                    className="checklist-form-status-and-dropdown"
                  >
                    <div
                      className="form-control checklist-form-status-div"
                    >
                      <i
                        className="glyphicon glyphicon-unchecked"
                        title="not started"
                      />
                    </div>
                    <select
                      className="form-control"
                      name="status"
                      onChange={[Function]}
                      value="NotStarted"
                    >
                      <option
                        value="NotStarted"
                      >
                        Not Started
                      </option>
                      <option
                        value="InProgress"
                      >
                        In Progress
                      </option>
                      <option
                        value="InProgressWithIssues"
                      >
                        In Progress With Issues
                      </option>
                      <option
                        value="CompletedSuccessfully"
                      >
                        Completed Successfully
                      </option>
                      <option
                        value="CompletedWithIssues"
                      >
                        Completed with Issues
                      </option>
                      <option
                        value="CompletedWithErrors"
                      >
                        Completed with Errors
                      </option>
                      <option
                        value="Canceled"
                      >
                        Canceled
                      </option>
                    </select>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="description"
              >
                Description
              </label>
              <div
                className="field"
              >
                <textarea
                  className="form-control"
                  name="description"
                  onChange={[Function]}
                  placeholder={undefined}
                  rows={2}
                  value="item description"
                />
              </div>
            </div>
            <div
              className="checklist-form-timestamps"
            >
              <div
                className="form-group"
              >
                <label
                  htmlFor="scheduledStartTime"
                >
                  Start Time: Scheduled
                </label>
                <div
                  className="field"
                >
                  <input
                    className="form-control"
                    name="scheduledStartTime"
                    onChange={[Function]}
                    placeholder={undefined}
                    type="text"
                    value="8:00 am"
                  />
                </div>
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="actualStartTime"
                >
                  Actual
                </label>
                <div
                  className="field"
                >
                  <input
                    className="form-control"
                    name="actualStartTime"
                    onChange={[Function]}
                    placeholder={undefined}
                    type="text"
                    value="8:03 am"
                  />
                </div>
              </div>
            </div>
            <div
              className="checklist-form-timestamps"
            >
              <div
                className="form-group"
              >
                <label
                  htmlFor="scheduledEndTime"
                >
                  End Time: Scheduled
                </label>
                <div
                  className="field"
                >
                  <input
                    className="form-control"
                    name="scheduledEndTime"
                    onChange={[Function]}
                    placeholder={undefined}
                    type="text"
                    value="9:00 am"
                  />
                </div>
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="actualEndTime"
                >
                  Actual
                </label>
                <div
                  className="field"
                >
                  <input
                    className="form-control"
                    name="actualEndTime"
                    onChange={[Function]}
                    placeholder={undefined}
                    type="text"
                    value="9:30 am"
                  />
                </div>
              </div>
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="userName"
              >
                Performed by
              </label>
              <div
                className="field"
              >
                <input
                  className="form-control"
                  name="userName"
                  onChange={[Function]}
                  placeholder={undefined}
                  type="text"
                  value="Jabba Script"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="modal-footer"
        >
          <div>
            <input
              className="btn btn-secondary pull-left"
              onClick={[Function]}
              type="button"
              value="Add Comment..."
            />
            <input
              className="btn btn-default"
              onClick={[Function]}
              type="button"
              value="Cancel"
            />
            <input
              className="btn btn-primary"
              onClick={[Function]}
              type="submit"
              value="Save"
            />
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`ChecklistItemEditorModal for new item should render correctly 1`] = `
<div
  className="checklist-chat-modal-backdrop"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog checklist-chat-modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="modal-header bg-primary text-white"
        >
          <button
            aria-label="Close"
            className="close"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
          <h4
            className="modal-title display-linebreaks"
          >
            Edit Checklist Item
          </h4>
        </div>
        <div
          className="modal-body"
        >
          <div>
            <div
              className="checklist-form-sequence-and-status-row"
            >
              <div
                className="form-group"
              >
                <label
                  htmlFor="sequenceNumber"
                >
                  Seq #
                </label>
                <div
                  className="field"
                >
                  <select
                    className="form-control"
                    name="sequenceNumber"
                    onChange={[Function]}
                    value="3"
                  >
                    <option
                      value="1"
                    >
                      1
                    </option>
                    <option
                      value="2"
                    >
                      2
                    </option>
                    <option
                      value="3"
                    >
                      3
                    </option>
                  </select>
                </div>
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="status"
                >
                  Status
                </label>
                <div
                  className="field"
                >
                  <div
                    className="checklist-form-status-and-dropdown"
                  >
                    <div
                      className="form-control checklist-form-status-div"
                    >
                      <i
                        className="glyphicon glyphicon-unchecked"
                        title="not started"
                      />
                    </div>
                    <select
                      className="form-control"
                      name="status"
                      onChange={[Function]}
                      value="NotStarted"
                    >
                      <option
                        value="NotStarted"
                      >
                        Not Started
                      </option>
                      <option
                        value="InProgress"
                      >
                        In Progress
                      </option>
                      <option
                        value="InProgressWithIssues"
                      >
                        In Progress With Issues
                      </option>
                      <option
                        value="CompletedSuccessfully"
                      >
                        Completed Successfully
                      </option>
                      <option
                        value="CompletedWithIssues"
                      >
                        Completed with Issues
                      </option>
                      <option
                        value="CompletedWithErrors"
                      >
                        Completed with Errors
                      </option>
                      <option
                        value="Canceled"
                      >
                        Canceled
                      </option>
                    </select>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="description"
              >
                Description
              </label>
              <div
                className="field"
              >
                <textarea
                  className="form-control"
                  name="description"
                  onChange={[Function]}
                  placeholder={undefined}
                  rows={2}
                  value=""
                />
              </div>
            </div>
            <div
              className="checklist-form-timestamps"
            >
              <div
                className="form-group"
              >
                <label
                  htmlFor="scheduledStartTime"
                >
                  Start Time: Scheduled
                </label>
                <div
                  className="field"
                >
                  <input
                    className="form-control"
                    name="scheduledStartTime"
                    onChange={[Function]}
                    placeholder={undefined}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="actualStartTime"
                >
                  Actual
                </label>
                <div
                  className="field"
                >
                  <input
                    className="form-control"
                    name="actualStartTime"
                    onChange={[Function]}
                    placeholder={undefined}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              className="checklist-form-timestamps"
            >
              <div
                className="form-group"
              >
                <label
                  htmlFor="scheduledEndTime"
                >
                  End Time: Scheduled
                </label>
                <div
                  className="field"
                >
                  <input
                    className="form-control"
                    name="scheduledEndTime"
                    onChange={[Function]}
                    placeholder={undefined}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="actualEndTime"
                >
                  Actual
                </label>
                <div
                  className="field"
                >
                  <input
                    className="form-control"
                    name="actualEndTime"
                    onChange={[Function]}
                    placeholder={undefined}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="userName"
              >
                Performed by
              </label>
              <div
                className="field"
              >
                <input
                  className="form-control"
                  name="userName"
                  onChange={[Function]}
                  placeholder={undefined}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="modal-footer"
        >
          <div>
            <input
              className="btn btn-secondary pull-left"
              onClick={[Function]}
              type="button"
              value="Add Comment..."
            />
            <input
              className="btn btn-default"
              onClick={[Function]}
              type="button"
              value="Cancel"
            />
            <input
              className="btn btn-primary"
              onClick={[Function]}
              type="submit"
              value="Save"
            />
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
